### **1. Generate a hollow inverted half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the hollow Inverted half pyramid pattern using numbers based on the given integer R.

**Sample Input :**

5

**Sample Output :**

12345

1  4

1 3

12

1

In [1]:
# Define the function hollow_inverted_half_pyramid which takes one argument, R
def hollow_inverted_half_pyramid(R):
    # Outer loop iterates from R down to 1 (inclusive), controlling the rows
    for i in range(R, 0, -1):
        # Inner loop iterates from 1 to i (inclusive), controlling the columns
        for j in range(1, i + 1):
            # Check if it's the first column, the last column, or the first row
            if j == 1 or j == i or i == R:
                # Print the value of j without moving to a new line
                print(j, end='')
            else:
                # Print a space character without moving to a new line
                print(' ', end='')
        # Move to the next line after finishing the current row
        print()

# Read an integer input from the user and store it in the variable R
R = int(input())

# Call the function hollow_inverted_half_pyramid with R as the argument
hollow_inverted_half_pyramid(R) 

12345
1  4
1 3
12
1


### **2. Write a code to generate a inverted half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input :**

5

**Sample Output :**

12345

1234

123

12

1

In [2]:
def inverted_half_pyramid(R):
    for i in range(R, 0, -1):
        for j in range(1, i + 1):
            print(j, end="")
        print()

R = int(input())
inverted_half_pyramid(R)

12345
1234
123
12
1


### **3. Write a code to generate a square pattern using the number '1'.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the square pattern with the number '1' in R*R form based on the given integer R.

**Sample Input :**

5

**Sample Output :**

11111

11111

11111

11111

11111

In [3]:
def square_pattern(R):
    for i in range(R):
        for j in range(R):
            print(1, end="")
        print()

# Sample Input
R = int(input("Enter the number of rows: "))
square_pattern(R)

11111
11111
11111
11111
11111


### **4. Write a code to generate a pyramid pattern using stars from the given input size N.**

#### **Input Description:**
Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**
Print the star pyramid with the given integer R.

**Sample Input :**

5

**Sample Output :**

*
* *
* * *
* * * *
* * * * *

In [2]:
n = int(input())

# Initialize an empty list to store the rows
pyramid = []

# Build the pyramid
for i in range(n):
    # Create a list of '*' for the current row
    row = ['*'] * (i + 1)
    # Join the list into a single string with spaces
    pyramid.append(' '.join(row))

# Print each row in the pyramid
for row in pyramid:
    print(row)

*
* *
* * *
* * * *
* * * * *


In [5]:
# Another way

def star_pyramid(N):
    for i in range(1, N + 1):
        for j in range(1, i + 1):
            print('*', end=' ')
        print()

N = int(input("Enter the number of rows: "))
star_pyramid(N)

* 
* * 
* * * 
* * * * 
* * * * * 


### **5. Generate a hollow half pyramid pattern using numbers.**

#### **Input Description:**
Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**
Print the hollow half pyramid pattern using numbers based on the given integer R.

**Sample Input:**

5

**Sample Output:**

1

12

1 3

1  4

12345

In [1]:
def hollow_half_pyramid(R):
    for i in range(1, R + 1):
        if i == 1:
            print("1")
        elif i == R:
            print("".join(str(x) for x in range(1, R + 1)))
        else:
            print("1" + " " * (i - 2) + str(i))

R = int(input())
hollow_half_pyramid(R)

1
12
1 3
1  4
12345


In [15]:
# Another way

def hollow_half_pyramid(R):
    for i in range(1, R + 1):
        for j in range(1, i + 1):
            if j == 1 or j == i or i == R:
                print(j, end="")
            else:
                print(" ", end="")
        print()
        
R = int(input())
hollow_half_pyramid(R)

1
12
1 3
1  4
12345


### **6. Write a code to generate a half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

5

**Sample Output :**

1

22

333

4444

55555

In [11]:
def number_half_pyramid(R):
    for i in range(1, R + 1):
        for j in range(i):
            print(i, end="")
        print()

R = int(input())
number_half_pyramid(R)

1
22
333
4444
55555


### **7. Write a code to generate a half pyramid number pattern.**

#### **Input Description:**

Given an even integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

5

**Sample Output:**

12345

4321

123

21

1

In [14]:
def pattern(n):
    for i in range(n):
        if i % 2 == 0:
            for j in range(n-i):
                print(j+1, end = "")
        else:
            for j in range(n-i, 0, -1):
                print(j, end = "")
        print()
        
n = int(input())
pattern(n)

12345
4321
123
21
1


### **8. Generate a solid rectangle using stars.**

#### **Input Description:**

Given an integer R indicates no of rows and an integer C indicates no of columns.

Where 1<=R<=100 and Where 1<=C<=100.

#### **Output Description:**

Print the rectangle using stars with R rows and C columns.

**Sample Input:**

3 5

**Sample Output:**

* * * * *
* * * * *
* * * * *

In [4]:
# Method - 1:

rows, columns = input().split()
rows = int(rows)
columns = int(columns)

for i in range(rows):
    for j in range(columns):
        if j != columns - 1:
            print("*", end=" ")
        else:
            print("*", end="")
    print("")

* * * * *
* * * * *
* * * * *


In [2]:
# Method - 2:

def solid_rectangle(R, C):
    for _ in range(R):
        print("* " * C)

R, C = map(int, input().split())            # R = 3, C = 5
solid_rectangle(R, C)


* * * * * 
* * * * * 
* * * * * 


In [12]:
# Method - 3:
rows, columns = map(int, input().split())

for i in range(rows):
    for j in range(columns):
        print("*", end=" ")
    print()

* * * * * 
* * * * * 
* * * * * 


### **9. Generate a half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the half pyramid pattern based on the given integer R.

**Sample Input:**

5

**Sample Output:**

1

12

123

1234

12345

In [16]:
n = int(input())            # n = 5

for i in range(1, n + 1):
    for j in range(1, i + 1):
        print(j, end="")
    print()

1
12
123
1234
12345


In [17]:
# Method 1: Using Nested Loops with String Concatenation

n = int(input("Enter the number of rows: "))

for i in range(1, n + 1):
    row = ""
    for j in range(1, i + 1):
        row += str(j)
    print(row)

# Method 2: Using List Comprehension and join()

n = int(input())

for i in range(1, n + 1):
    print("".join(str(j) for j in range(1, i + 1)))

# Method 3: Using *args in print

n = int(input("Enter the number of rows: "))

for i in range(1, n + 1):
    print(*range(1, i + 1), sep="")

# Method 4: Using a While Loop

n = int(input("Enter the number of rows: "))

i = 1
while i <= n:
    j = 1
    while j <= i:
        print(j, end="")
        j += 1
    print()
    i += 1

# Method 5: Using Recursion

def print_row(row):
    if row > 0:
        print_row(row - 1)
        print("".join(str(i) for i in range(1, row + 1)))

n = int(input("Enter the number of rows: "))
print_row(n)

1
12
123
1234
12345
1
12
123
1234
12345
1
12
123
1234
12345
1
12
123
1234
12345
1
12
123
1234
12345


### **10. Generate a  full pyramid using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the full pyramid using numbers based on the given integer R.

**Sample Input:**

5

**Sample Output:**

    1

   232

  34543

 4567654

567898765

In [18]:
def full_pyramid(R):
    for i in range(1, R + 1):
        # Print leading spaces
        for j in range(R - i):
            print(" ", end="")
        
        # Print increasing numbers
        for j in range(i, 2 * i):
            print(j, end="")
        
        # Print decreasing numbers
        for j in range(2 * i - 2, i - 1, -1):
            print(j, end="")
        
        # Move to the next line
        print()

R = int(input())
full_pyramid(R)

    1
   232
  34543
 4567654
567898765


#### **Logic Explanation:**

def full_pyramid(R):
    Defines a function full_pyramid that takes an integer R as input to determine the number of rows in the pyramid.

Leading Spaces:
    The first inner loop (for j in range(R - i)) prints spaces to center-align the pyramid.

Increasing Numbers:
    The second inner loop (for j in range(1, i + 1)) prints numbers in increasing order from 1 to i.

Decreasing Numbers:
    The third inner loop (for j in range(i - 1, 0, -1)) prints numbers in decreasing order from i-1 down to 1.

print():
    Moves to the next line after printing each row to start a new row.

In [19]:
def full_pyramid(R):
    for i in range(1, R + 1):
        # Print leading spaces
        for j in range(R - i):
            print(" ", end="")
        
        # Print numbers in increasing order
        for j in range(1, i + 1):
            print(j, end="")
        
        # Print numbers in decreasing order
        for j in range(i - 1, 0, -1):
            print(j, end="")
        
        # Move to the next line for the next row
        print()

R = int(input("Enter the number of rows: "))
full_pyramid(R)

    1
   121
  12321
 1234321
123454321


### **11. Generate a hollow inverted half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the hollow inverted half pyramid pattern using numbers based on the given integer R.

**Sample Input:**

5

**Sample Output:**

    1

   1 2

  1   3

 1     4

1 2 3 4 5

In [21]:
n=int(input())

for i in range(n):
  p=1
  l=[]

  # Add spaces for alignment
  for j in range(i, n-1):
    # print("",end=" ")
    l.append("")

  # Fill the pattern with numbers and spaces
  for j in range(i+1):
    if i==n-1 or j==0 or j==i:
      # print (p,end=" ")
      l.append(p)
    else:
      #print(" ",end=" ")
      l.append(" ")
      #print(*l)
    p=p+1
  #print()
  print(*l)

    1
   1 2
  1   3
 1     4
1 2 3 4 5


In [38]:
x = int(input("Enter the number of rows (x): "))
row = 1
while row <= x:
    # Print spaces to align the pyramid
    space = x - row
    while space > 0:
        print(" ", end="")
        space -= 1
    
    # Print numbers for the current row
    column = 1
    while column <= row:
        print(column, end=" ")
        column += 1
    
    # Move to the next row
    print()  # Move to the next line after printing the row
    row += 1

    1 
   1 2 
  1 2 3 
 1 2 3 4 
1 2 3 4 5 


In [37]:
n = int(input())

for i in range(n):
    p = 1
    l = []
    
    # Add spaces for alignment
    for j in range(i, n - 1):
        l.append(" ")
    
    # Fill the pattern with numbers and spaces
    for j in range(i + 1):
        if i == n - 1 or j == 0 or j == i:
            l.append(str(p))
        else:
            l.append(" ")
        p += 1
    print(*l)
    # Print the row
    # print("".join(l))

        1
      1 2
    1   3
  1     4
1 2 3 4 5


### **12. Write a code to generate the following pattern.**

#### **Input Description:**

Input consists of a single integer that corresponds to R, the number of rows. R is always an odd number.

Where 1<=R<=100.

#### **Output Description:**

Print the character with the letter 'b' pattern with the size based on the given integer R.

**Sample Input:**

5

**Sample Output:**

*********

b*******b

bb*****bb

bbb***bbb

bbbb*bbbb

In [30]:
n = int(input("Enter the value of n: "))
for i in range(n):
    for j in range(i):
        print("b", end="")
    for j in range(i, n-1):
        print("*", end="")
    for j in range(i, n):
        print("*", end="")
    for j in range(i):
        print("b", end="")
    print()

*********
b*******b
bb*****bb
bbb***bbb
bbbb*bbbb


### **13. Write a code to generate a pyramid using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the pyramid pattern using stars based on the given integer R.

**Sample Input:**

6

**Sample Output:**

     *

    **

   ***

  ****

 *****

******

In [24]:
def pyramid(R):
    for i in range(1, R + 1):
        # Calculate number of spaces and stars for each row
        num_spaces = R - i
        num_stars = i
        
        # Construct the row string
        row = ' ' * num_spaces + '*' * num_stars
        print(row)

R = int(input())        # R = 6
pyramid(R)

     *
    **
   ***
  ****
 *****
******


### **14. Write a code to generate a hollow rectangle using stars.**

#### **Input Description:**

Given an integer R indicates no of rows and an integer C indicates no of columns.

Where 1<=R<=100 and Where 1<=C<=100.

#### **Output Description:**

Print the hollow rectangle using stars with R rows and C columns.

**Sample Input :**

3 5

**Sample Output:**

* * * * *

*       *

* * * * *

In [40]:
def pattern(r, c):
    for i in range(1, r+1):
        l = []
        if i == 1 or i == r:
            for j in range(c):
                l.append('*')
        else:
            l.append('*')
            for j in range(1, c-1):
                l.append(' ')
            l.append('*')
        print(*l)

r, c = map(int, input().split())
pattern(r, c)

* * * * *
*       *
* * * * *


In [42]:
# Method - 1:
n, k = map(int, input().split())

if 1 <= n <= 100 and 1 <= k <= 100:
    # Print the first row (all stars)
    print("* " * k)
    
    # Print middle rows
    for row in range(1, n - 1):
        print("*", end=" ")
        for col in range(1, k - 1):
            print(" ", end=" ")
        print("*")
    
    # Print the last row (all stars)
    if n > 1:
        print("* " * k)

# Method - 2:
n, k = map(int, input().split())

if 1 <= n <= 100 and 1 <= k <= 100:
    for row in range(1, n + 1):
        for col in range(1, k + 1):
            if col == 1 or col == k or row == 1 or row == n:
                print("*", end=" ")
            else:
                print(" ", end=" ")
        print()

# Method - 3:
def hollow_rectangle(R, C):
    for i in range(R):
        if i == 0 or i == R - 1:  # First and last rows
            print("* " * C)
        else:  # Middle rows
            print("*" + " " * (2 * C - 3) + "*")

# Reading input
R, C = map(int, input().split())
hollow_rectangle(R, C)

* * * * * 
*       *
* * * * * 
* * * * * 
*       * 
* * * * * 
* * * * * 
*       *
* * * * * 


### **15. Write a code to generate a solid half diamond pattern using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the solid half diamond pattern based on the given integer R.

**Sample Input:**

5

**Sample Output:**

*
**
***
****
*****
****
***
**
*

In [43]:
def solid_half_diamond(R):
    # Upper half of the diamond
    for i in range(1, R + 1):
        print("*" * i)
    
    # Lower half of the diamond (excluding the middle line)
    for i in range(R - 1, 0, -1):
        print("*" * i)

R = int(input())

# Validating input range
if 1 <= R <= 100:
    solid_half_diamond(R)
else:
    print("Input out of range. Please enter a number between 1 and 100.")

*
**
***
****
*****
****
***
**
*


### **16. Write a code to generate a full pyramid pattern using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the star pyramid with the given integer R.

**Sample Input:**

5

**Sample Output:**
    *
   * *
  * * *
 * * * *
* * * * *

In [44]:
def full_pyramid(R):
    for i in range(1, R + 1):
        # Print leading spaces
        print(" " * (R - i), end="")
        
        # Print stars for current row
        for j in range(1, i + 1):
            if j == 1:
                print("*", end="")
            else:
                print(" *", end="")
        
        # Move to the next line after printing the stars for the current row
        print()

R = int(input())

# Validating input range
if 1 <= R <= 100:
    full_pyramid(R)
else:
    print("Input out of range. Please enter a number between 1 and 100.")

    *
   * *
  * * *
 * * * *
* * * * *


### **17. Write a code to generate the following pattern.**

#### **Input Description:**

Given an odd integer R indicates number of rows.R is always an odd number.

where 1<=R<=100.

#### **Output Description:**

Print the character pattern based on the given input R.

**Sample Input:**

9

**Sample Output:**

bbbb*bbbb

bbb***bbb

bb*****bb

b*******b

*********

b*******b

bb*****bb

bbb***bbb

bbbb*bbbb

In [96]:
def diamond_pattern(rows):
    for i in range(1, rows):
        # Printing spaces
        for j in range(rows - i):
            print("b", end="")
        # Printing stars
        for k in range(1, 2 * i):
            print("*", end="")
        # Printing spaces
        for j in range(rows - i):
            print("b", end="")
        print()

    # Middle line (full stars)
    for i in range(rows * 2 - 1):
        print("*", end="")
    print()

    # Lower part (symmetric to upper part)
    for i in range(rows - 1, 0, -1):
        # Printing spaces
        for j in range(rows - i):
            print("b", end="")
        # Printing stars
        for k in range(1, 2 * i):
            print("*", end="")
        # Printing spaces
        for j in range(rows - i):
            print("b", end="")
        print()

num_rows = int(input())     # num_rows = 5
diamond_pattern(num_rows)

bbbb*bbbb
bbb***bbb
bb*****bb
b*******b
*********
b*******b
bb*****bb
bbb***bbb
bbbb*bbbb


### **18. Write a code to generate an inverted half pyramid pattern using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the star inverted pyramid with the given integer R.

**Sample Input:**

5

**Sample Output:**

*  *  *  *  *
*  *  *  *
*  *  *
*  *
*

In [41]:
def pattern(n):
  for i in range(n):
    l=[]
    for j in range(n-i):
      l.append("*")
    print('  '.join(l))

n = int(input())
pattern(n)

*  *  *  *  *
*  *  *  *
*  *  *
*  *
*


In [39]:
def reverse_pattern(rows):
    for i in range(rows, 0, -1):
        for j in range(i):
            print("*", end=" ")
        print()

num_rows = int(input())
reverse_pattern(num_rows)

print("\n")

# Another way:
def inverted_half_pyramid(R):
    for i in range(R, 0, -1):
        print("* " * i)

R = int(input())

# Validating input range
if 1 <= R <= 100:
    inverted_half_pyramid(R)
else:
    print("Input out of range. Please enter a number between 1 and 100.")

* * * * * 
* * * * 
* * * 
* * 
* 


* * * * * 
* * * * 
* * * 
* * 
* 


### **19. Generate a number pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

4

**Sample Output:**

1234567

12345

123

1

In [48]:
# Get the number of rows as input
R = int(input())

# Generate the pyramid pattern
for row in range(R, 0, -1):
    # Print the numbers in each row
    for num in range(1, 2 * row):
        print(num, end="")
    print()

1234567
12345
123
1


### **20. Write a code to generate a half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

5

**Sample Output:**

12345

2345

345

45

5

In [46]:
def number_half_pyramid(R):
    for i in range(R):
        for j in range(i + 1, R + 1):
            print(j, end="")
        print()

R = int(input())
if 1 <= R <= 100:
    number_half_pyramid(R)
else:
    print("Input out of range. Please enter a number between 1 and 100.")

12345
2345
345
45
5


### **21. Generate a floyd's triangle.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print a floyd's triangle based on the given integer R.

**Sample Input:**

5

**Sample Output:**

1

2 3

4 5 6

7 8 9 10

11 12 13 14 15

In [38]:
def generate_number_triangle(rows):
    current_number = 1
    for i in range(1, rows + 1):
        row_numbers = []
        for j in range(i):
            row_numbers.append(current_number)
            current_number += 1
        print(" ".join(map(str, row_numbers)))

rows = int(input())
generate_number_triangle(rows)

1
2 3
4 5 6
7 8 9 10
11 12 13 14 15


In [37]:
def floyds_triangle(R):
    num = 1
    for i in range(1, R + 1):
        for j in range(1, i + 1):
            print(num, end=" ")
            num += 1
        print()

R = int(input())
floyds_triangle(R)

print("\n")

rows = int(input())         # rows = 5
num = 1 
sum = 0 
for i in range(rows): 
    for j in range(i + 1, rows+1): 
        print(num, end=' ')
        sum = sum + num 
        num = num + 1     
    print() 
print("\nSum of numbers: ", sum) 

1 
2 3 
4 5 6 
7 8 9 10 
11 12 13 14 15 


1 2 3 4 5 
6 7 8 9 
10 11 12 
13 14 
15 

Sum of numbers:  120


### **22. Write a code to generate a half pyramid pattern using number.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

5

**Sample Output:**

54321

4321

321

21

1

In [89]:
def half_pyramid_pattern(R):
    for i in range(R, 0, -1):
        for j in range(i, 0, -1):
            print(j, end='')
        print()

R = int(input())
half_pyramid_pattern(R)

54321
4321
321
21
1


### **23. Write a code to generate a hollow full pyramid pattern using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the star hollow full pyramid with the given integer R.

**Sample Input:**

5

**Sample Output:**

    *
   * *
  *   *
 *     *
* * * * *

In [42]:
rows = 5

# Loop through each row
for i in range(rows):
    # Print leading spaces based on the current row
    for j in range(rows - i - 1):
        print(" ", end="")

    # Print asterisks based on the current row (i + 1)
    for j in range(i + 1):
        if j == 0 or j == i or i == rows - 1:
            print("*", end=" ")
        else:
            print(" ", end=" ")

    # Move to the next line
    print()

    * 
   * * 
  *   * 
 *     * 
* * * * * 


In [44]:
# Similar patten

rows = 5

# Loop through each row
for i in range(rows):
  # Print leading spaces based on the current row
  for j in range(rows - i - 1):
    print(" ", end="")

  # Print asterisks based on the current row (i + 1)
  for j in range(i + 1):
    print("*", end=" ")

  # Move to the next line
  print()

def hollow_full_pyramid(R):
    for i in range(R):
        # Print leading spaces
        for j in range(R - i - 1):
            print(" ", end=" ")

        # Print stars and inner spaces
        for j in range(i + 1):
            if j == 0 or j == i or i == R - 1:
                print("*", end=" ")
            else:
                print(" ", end=" ")

        # Move to the next line after each row
        print()

n = 5
hollow_full_pyramid(n)

    * 
   * * 
  * * * 
 * * * * 
* * * * * 
        * 
      * * 
    *   * 
  *     * 
* * * * * 


### **24. Write a code to generate a half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

5

**Sample Output:**

55555

4444

333

22

1

In [88]:
def half_pyramid_numbers(R):
    for i in range(R, 0, -1):
        for j in range(i):
            print(i, end='')
        print()

R = int(input())
half_pyramid_numbers(R)

55555
4444
333
22
1


### **25. Write a code to generate a aplhabet half pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=26.

#### **Output Description:**

Print the alphabet half pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

EDCBA

EDCB

EDC

ED

E

In [159]:
def generate_alphabet_pyramid(rows):
    if rows < 1 or rows > 26:
        print("Invalid input. R must be between 1 and 26.")
        return
    
    base_char = ord('A') + rows - 1
    
    for i in range(rows):
        for j in range(base_char, ord('A') + i - 1, -1):
            print(chr(j), end="")
        print()

try:
    R = int(input())
    generate_alphabet_pyramid(R)
except ValueError:
    print("Invalid input. Please enter a valid integer.")

EDCBA
EDCB
EDC
ED
E


### **26. In the IPL season’s valedictory function the organizers have organized for a dance program. The dance has to be performed by men along with the points of the diagonals of the square of side ‘n’ and the females along with points of the borders. The remaining positions are filled by children. You have to determine their respective positions by writing a program.**

#### **Input Description:**

Given an integer N indicates representing the matrix (N*N).

where 1<=N<=100.

#### **Output Description:**

Print the N*N character matrix with the character F(Female), M(Male), C(Children).

Sample Input :

7

Sample Output :

M F F F F F M

F M C C C M F

F C M C M C F

F C C M C C F

F C M C M C F

F M C C C M F

M F F F F F M

In [160]:
def generate_dance_matrix(N):
    if N < 1 or N > 100:
        print("Invalid input. N must be between 1 and 100.")
        return
    
    matrix = [['C' for _ in range(N)] for _ in range(N)]
    
    # Fill the matrix based on the rules
    for i in range(N):
        for j in range(N):
            if i == 0 or i == N - 1 or j == 0 or j == N - 1:
                matrix[i][j] = 'F'
            if i == j or i + j == N - 1:
                matrix[i][j] = 'M'
    
    # Print the matrix
    for row in matrix:
        print(' '.join(row))

# Taking input from the user
try:
    N = int(input("Enter the size of the matrix (N): "))
    generate_dance_matrix(N)
except ValueError:
    print("Invalid input. Please enter a valid integer.")

M F F F F F M
F M C C C M F
F C M C M C F
F C C M C C F
F C M C M C F
F M C C C M F
M F F F F F M


### **27. Generate the following inverted character with star pattern.**

#### **Input Description:**

Input consists of a single integer that corresponds to R, the number of rows. R is always an odd number. where 1<=R<=100.

#### **Output Description:**

Print the inverted character pattern from the given input size R.

**Sample Input:**

5

**Sample Output:**

bbbb*bbbb
bbb***bbb
bb*****bb
b*******b
*********


In [36]:
n = int(input())

for i in range(n):
  for j in range(i,n-1):
    print("b", end="")

  for j in range (i):
    print("*", end="")

  for j in range (i+1):
    print("*", end="")

  for j in range(i,n-1):
    print("b", end="")
  print()

bbbb*bbbb
bbb***bbb
bb*****bb
b*******b
*********


### **28. Write a code to generate an inverted full pyramid pattern using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the star inverted full pyramid with the given integer R.

**Sample Input:**

5

**Sample Output:**

* * * * *
 * * * *
  * * *
   * *
    *

In [172]:
def inverted_full_pyramid(rows):
    # Outer loop for rows
    for i in range(rows, 0, -1):
        # Print spaces before stars
        for j in range(rows - i):
            print(" ", end="")
        
        # Print stars
        for k in range(i):
            if k == i - 1:
                print("*", end="")
            else:
                print("* ", end="")
        print()

rows = int(input())
inverted_full_pyramid(rows)

* * * * *
 * * * *
  * * *
   * *
    *


### **29. Write a code to generate a pyramid using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the pyramid pattern using stars based on the given integer R.

**Sample Input:**

5

**Sample Output:**

*****

 ****

  ***

   **

    *

In [2]:
def generate_pyramid(R):
    for i in range(R, 0, -1):
        # Print spaces
        for j in range(R - i):
            print(" ", end="")
        # Print stars
        for j in range(i):
            print("*", end="")
        print()

R = int(input())
generate_pyramid(R)

*****
 ****
  ***
   **
    *


### **30. Write a code to generate an Rhombus Pattern using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the solid rhombus using stars with the size of rhombus R.

**Sample Input:**

4

**Sample Output:**

   ****
  ****
 ****
****

In [3]:
def generate_rhombus(R):
    for i in range(R):
        # Print spaces
        for j in range(R - i - 1):
            print(" ", end="")
        # Print stars
        for j in range(R):
            print("*", end="")
        print()
        
R = int(input())
generate_rhombus(R)

   ****
  ****
 ****
****


### **31. Write a code to generate a aplhabet pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=26.

#### **Output Description:**

Print the alphabet pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

    A

   ABA

  ABCAB

 ABCDABC

ABCDEABCD

In [49]:
def pattern(n):
    c = 65
    for i in range(1, n+1):
        for j in range(n-i):
            print(' ', end = "")
        for j in range(i):
            print(chr(c+j), end = "")
        for j in range(i-1):
            print(chr(c+j), end = "")
        print()
        
n = int(input())
pattern(n)

    A
   ABA
  ABCAB
 ABCDABC
ABCDEABCD


### **32. Write a code to generate a half pyramid number pattern.**

#### **Input Description:**

Given an even integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number pattern based on the given integer R.

**Sample Input:**

5

**Sample Output:**

12345

1234

123

12

1

In [47]:
def half_pyramid_numbers(R):
    # Loop through each row
    for i in range(R, 0, -1):
        # Print numbers from 1 to i
        for j in range(1, i + 1):
            print(j, end="")
        print()

R = int(input())
half_pyramid_numbers(R)

12345
1234
123
12
1


### **33. Write a code to generate a half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

5

**Sample Output:**

13579

3579

579

79

9

In [50]:
def pattern(n):
    for i in range(1, (2*n), 2):
        l = []
        for j in range(i, 2*n, 2):
            l.append(str(j))
        print(''.join(l))
    
n = int(input())
pattern(n)

13579
3579
579
79
9


### **34. Generate a half diamond pattern using stars and numbers in a palindromic pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the half diamond pattern using stars and numbers in a palindromic pattern based on the given integer R.

**Sample Input:**

3

**Sample Output:**

*
*1*
*121*
*12321*
*121*
*1*
*

In [37]:
def palindrome_pattern(n):
    # Print the first asterisk
    print('*')
    
    # Upper half of the pattern
    for i in range(1, n + 1):
        l = ['*']  # Start each line with an asterisk
        # Append increasing numbers
        for j in range(1, i + 1):
            l.append(str(j))
        # Append decreasing numbers
        for j in range(i - 1, 0, -1):
            l.append(str(j))
        l.append('*')  # End each line with an asterisk
        print(''.join(l))  # Join the list into a string and print it
    
    # Lower half of the pattern
    for i in range(n - 1, 0, -1):
        l = ['*']  # Start each line with an asterisk
        # Append increasing numbers
        for j in range(1, i + 1):
            l.append(str(j))
        # Append decreasing numbers
        for j in range(i - 1, 0, -1):
            l.append(str(j))
        l.append('*')  # End each line with an asterisk
        print(''.join(l))  # Join the list into a string and print it
    
    # Print the last asterisk
    print('*')

# Read input and generate the pattern
n = int(input("Enter the value of n: "))
palindrome_pattern(n)

*
*1*
*121*
*12321*
*121*
*1*
*


### **35. Generate the aplhabet pattern using nested loops.**

#### **Input Description:**

Input consists of a string S.where length of the string (S)

Where 2<=S<=20.

#### **Output Description:**

Print the alphabet pattern from the given input string S.

**Sample Input:**

abcdef

**Sample Output:**

abcdef
b    e
c    d
d    c
e    b
fedcba

In [55]:
def alphabet_pattern(S):
    length = len(S)
    if length < 2 or length > 20:
        print("Invalid input. Length of S should be between 2 and 20.")
        return
    
    # Print the first line (original string)
    print(S)
    
    # Print the middle lines
    for i in range(1, length - 1):
        line = S[i] + " " * (length - 2) + S[length - 1 - i]
        print(line)
    
    # Print the last line (reversed string)
    print(S[::-1])

S = input()
alphabet_pattern(S)

abcdef
b    e
c    d
d    c
e    b
fedcba


### **36. Write a code to generate a aplhabet pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=26.

#### **Output Description:**

Print the alphabet pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

    A
   
   BAB
  
  CBABC
 
 DCBABCD

EDCBABCDE

In [68]:
def pattern(n):
    c = 65  # ASCII value for 'A'
    for i in range(1, n + 1):
        # Print leading spaces
        for j in range(n - i):
            print(' ', end="")
        
        # Print decreasing part of the pattern
        for j in range(i - 1, -1, -1):
            print(chr(c + j), end="")
        
        # Print increasing part of the pattern
        for j in range(1, i):
            print(chr(c + j), end="")
        print()

n = int(input())
pattern(n)

    A
   BAB
  CBABC
 DCBABCD
EDCBABCDE


### **37. Write a code to generate a right arrow using patterns.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the right arrow pattern using stars based on the given integer R.

**Sample Input:**

5

**Sample Output:**

    *
   *
  *
 *
*****
 *
  *
   *
    *

In [38]:
# Method - 1:

def pattern(n):
    for i in range(n-1):
        for j in range(n-i-1):
            print(' ', end = "")
        print('*')
    print('*' * n)
    for i in range(n-2, -1, -1):
        for j in range(n-i-1):
            print(' ', end = "")
        print('*')

n = int(input())
pattern(n)

    *
   *
  *
 *
*****
 *
  *
   *
    *


In [64]:
# Method - 2:

def right_arrow_pattern(R):
    if R < 1 or R > 100:
        print("Invalid input. R should be between 1 and 100.")
        return
    
    # Upper part of the arrow
    for i in range(1, R):
        # Print leading spaces
        print(" " * (R - i - 1), end="")
        print("*")
    
    # Middle part of the arrow
    print("*" * R)
    
    # Lower part of the arrow
    for i in range(1, R):
        # Print leading spaces
        print(" " * i, end="")
        print("*")

R = int(input())
right_arrow_pattern(R)

   *
  *
 *
*
*****
 *
  *
   *
    *


### **38. Write a code to generate a left arrow pattern using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the left arrow pattern based on the given integer R.

**Sample Input:**

5

**Sample Output:**

*
 *
  *
   *
*****
   *
  *
 *
*

In [39]:
# Method - 1:

def pattern(n):
    for i in range(n-1):
        for j in range(i):
            print(' ', end = "")
        print('*')
    print('*' * n)
    for i in range(n-2, -1, -1):
        for j in range(i):
            print(' ', end = "")
        print('*')

n = int(input())
pattern(n)

*
 *
  *
   *
*****
   *
  *
 *
*


In [65]:
def left_arrow_pattern(R):
    if R < 1 or R > 100:
        print("Invalid input. R should be between 1 and 100.")
        return
    
    # Upper part of the arrow
    for i in range(R):
        # Print leading spaces
        print(" " * i, end="")
        
        # Print star
        print("*")
    
    # Middle part of the arrow
    print("*" * R)
    
    # Lower part of the arrow
    for i in range(R - 1, -1, -1):
        # Print leading spaces
        print(" " * i, end="")
        
        # Print star
        print("*")

R = int(input())
left_arrow_pattern(R)

*
 *
  *
   *
    *
*****
    *
   *
  *
 *
*


### **39. Write a code to generate a pyramid pattern on numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the pyramid number pattern based on the given integer R.

**Sample Input:**

5

**Sample Output:**

    0
   
   101
  
  21012
 
 3210123

432101234

In [66]:
def number_pyramid(R):
    if R < 1 or R > 100:
        print("Invalid input. R should be between 1 and 100.")
        return
    
    # Loop through each row
    for i in range(R):
        # Print leading spaces
        print(" " * (R - i - 1), end="")
        
        # Print numbers in decreasing order
        for j in range(i, -1, -1):
            print(j, end="")
        
        # Print numbers in increasing order (excluding the first zero)
        for j in range(1, i + 1):
            print(j, end="")
        print()

R = int(input())
number_pyramid(R)

    0
   101
  21012
 3210123
432101234


### **40. Write a code to generate a half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

5

**Sample Output:**

1

24

135

2468

13579

In [67]:
def number_half_pyramid(R):
    if R < 1 or R > 100:
        print("Invalid input. R should be between 1 and 100.")
        return
    
    # Loop through each row
    for i in range(1, R + 1):
        # Check if the row is even or odd
        if i % 2 == 0:
            # Even row: print even numbers
            for j in range(1, i + 1):
                print(j * 2, end="")
        else:
            # Odd row: print odd numbers
            for j in range(i):
                print(j * 2 + 1, end="")
        
        # Move to the next line after each row is printed
        print()

R = int(input())
number_half_pyramid(R)

1
24
135
2468
13579


### **41. Write a code to generate a aplhabet half pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=26.

#### **Output Description:**

Print the alphabet half pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

ABCDE

 ABCD

  ABC

   AB

    A

In [69]:
def alphabet_half_pyramid(R):
    if R < 1 or R > 26:
        print("Invalid input. R should be between 1 and 26.")
        return
    
    c = 65  # ASCII value for 'A'
    
    for i in range(R):
        # Print leading spaces
        print(" " * i, end="")
        
        # Print the alphabets
        for j in range(R - i):
            print(chr(c + j), end="")
        
        # Move to the next line after each row is printed
        print()

R = int(input())
alphabet_half_pyramid(R)

ABCDE
 ABCD
  ABC
   AB
    A


### **42. Write a code to generate a half pyramid pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number half pyramid pattern with the size R.

**Sample Input:**

5

**Sample Output:**

5

45

345

2345

12345

In [70]:
def generate_half_pyramid(R):
    for i in range(1, R + 1):
        for j in range(R - i + 1, R + 1):
            print(j, end="")
        print()

R = int(input())
generate_half_pyramid(R)

5
45
345
2345
12345


### **43. Write a code to generate an hollow rhombus Pattern using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the solid rhombus using stars with the size R.

**Sample Input:**

4

**Sample Output:**

   ****
  *  *
 *  *
****

In [71]:
def generate_hollow_rhombus(R):
    for i in range(1, R + 1):
        # Print leading spaces
        for j in range(R - i):
            print(" ", end="")
        # Print the stars and spaces for the hollow rhombus
        for j in range(1, R + 1):
            if i == 1 or i == R or j == 1 or j == R:
                print("*", end="")
            else:
                print(" ", end="")
        print()

R = int(input())
generate_hollow_rhombus(R)

   ****
  *  *
 *  *
****


### **44. Write a code to generate the following pattern.**

#### **Input Description:**

Given an even integer R indicates number of stars in first and last row.R is always an even number.

Where 2<=R<=100.

#### **Output Description:**

Print the pattern based on the given integer R.

**Sample Input:**

10

**Sample Output:**

**********

****bb****

***bbbb***

**bbbbbb**

*bbbbbbbb*

**bbbbbb**

***bbbb***

****bb****

**********

In [35]:
n = int(input())
n = n // 2

# Upper half of the pattern
for i in range(n - 1):
    # Print the first set of asterisks
    for j in range(i, n):
        print("*", end="")
    # Print the first set of 'b'
    for j in range(i):
        print("b", end="")
    # Print the second set of 'b'
    for j in range(i):
        print("b", end="")
    # Print the second set of asterisks
    for j in range(i, n):
        print("*", end="")
    # Move to the next line
    print()

# Lower half of the pattern
for i in range(n):
    # Print the first set of asterisks
    for j in range(i + 1):
        print("*", end="")
    # Print the first set of 'b'
    for j in range(i, n - 1):
        print("b", end="")
    # Print the second set of 'b'
    for j in range(i + 1, n):
        print("b", end="")
    # Print the second set of asterisks
    for j in range(i + 1):
        print("*", end="")
    # Move to the next line
    print()

**********
****bb****
***bbbb***
**bbbbbb**
*bbbbbbbb*
**bbbbbb**
***bbbb***
****bb****
**********


### **45. Write a code to generate a square pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of R*2-1 rows.

Where 1<=R<=100.

#### **Output Description:**

Print the square pyramid number based on the given integer R.

**Sample Input:**

5

**Sample Output:**

1

1 4

1 4 9

1 4 9 16

1 4 9 16 25

1 4 9 16

1 4 9

1 4

1

In [73]:
# Method - 1:
def pattern(n):
    for i in range(1, n+1):
        l = []
        for j in range(1, i+1):
            l.append(j**2)
        print(*l)
    for i in range(n-1, 0, -1):
        l = []
        for j in range(1, i+1):
            l.append(j**2)
        print(*l)
n = int(input())
pattern(n)

1
1 4
1 4 9
1 4 9 16
1 4 9 16 25
1 4 9 16
1 4 9
1 4
1


In [72]:
# Method - 2:
def generate_square_pattern(R):
    # Upper part of the pattern
    for i in range(1, R + 1):
        for j in range(1, i + 1):
            print(j * j, end=" ")
        print()
    
    # Lower part of the pattern
    for i in range(R - 1, 0, -1):
        for j in range(1, i + 1):
            print(j * j, end=" ")
        print()

R = int(input())
generate_square_pattern(R)

1 
1 4 
1 4 9 
1 4 9 16 
1 4 9 16 25 
1 4 9 16 
1 4 9 
1 4 
1 


### **46. Write a code to generate a pyramid pattern on numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the pyramid number pattern based on the given integer R.

**Sample Input:**

5

**Sample Output:**

999999999

 7777777

  55555

   333

    1

In [1]:
# Method - 1:

n = int(input())  # Read the number of rows
p = (n * 2) - 1  # Calculate the starting number

for i in range(n):  # Loop for each row
    # Print leading spaces
    for j in range(i):
        print(" ", end="")

    # Print the number `p` for the remaining columns in the row
    for j in range(i, n - 1):
        print(p, end="")

    for j in range(i, n):
        print(p, end="")

    # Decrement the number to be printed in the next row
    p -= 2
    print()

999999999
 7777777
  55555
   333
    1


In [75]:
# Method - 2:

def generate_number_pyramid(R):
    current_number = 9
    for i in range(R, 0, -1):
        # Print leading spaces
        for _ in range(R - i):
            print(" ", end="")
        # Print numbers
        for _ in range(2 * i - 1):
            print(current_number, end="")
        print()
        current_number -= 2

R = int(input())
generate_number_pyramid(R)

999999999
 7777777
  55555
   333
    1


### **47. Write a code to generate a aplhabet pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=26.

#### **Output Description:**

Print the alphabet pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

    A

   BBB

  CCCCC

 DDDDDDD

EEEEEEEEE

In [2]:
def pattern(n):
    c = 65  # ASCII value for 'A'
    for i in range(n):
        # Print leading spaces
        for j in range(n - i - 1):
            print(' ', end="")
        
        # Print the current alphabet character (2*i + 1) times
        for j in range((2 * i) + 1):
            print(chr(c + i), end="")
        
        # Move to the next line
        print()

n = int(input())
pattern(n)

    A
   BBB
  CCCCC
 DDDDDDD
EEEEEEEEE


### **48. Write a code to generate a half pyramid number pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number half pyramid pattern based on the given integer R.

**Sample Input:**

5

**Sample Output:**

1

2 4

1 3 5

2 4 6 8

1 3 5 7 9

In [3]:
def pattern(n):
    o = ['1']
    e = ['2']
    for i in range(1, n+1):
        if i % 2 == 1:
            print(*o)
        else:
            print(*e)
        o.append(str(int(o[-1]) + 2))
        e.append(str(int(e[-1]) + 2))
            
n = int(input())
pattern(n)

1
2 4
1 3 5
2 4 6 8
1 3 5 7 9


In [4]:
def half_pyramid_number_pattern(R):
    for i in range(1, R + 1):
        if i % 2 == 1:
            # Odd row: Print odd numbers
            for j in range(i):
                print(2 * j + 1, end=" ")
        else:
            # Even row: Print even numbers
            for j in range(1, i + 1):
                print(2 * j, end=" ")
        print()

R = int(input())
half_pyramid_number_pattern(R)

1 
2 4 
1 3 5 
2 4 6 8 
1 3 5 7 9 


### **49. Write a code to generate a alphabet pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=26.

#### **Output Description:**

Print the alphabet pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

    A

   ABC

  ABCDE

 ABCDEFG

ABCDEFGHI

In [6]:
# Method - 1:

n = int(input())

for i in range(n):
  p=65
  for j in range(i+1, n):
    print(" ", end="")
  
  for j in range (i):
    print(chr(p), end="")
    p=p+1
  for j in range (i+1):
    print(chr(p), end="")
    p=p+1
  print()

    A
   ABC
  ABCDE
 ABCDEFG
ABCDEFGHI


In [5]:
# Method - 2:

def alphabet_pyramid(R):
    for i in range(R):
        # Print leading spaces
        for j in range(R - i - 1):
            print(' ', end='')
        
        # Print alphabet characters
        for j in range(2 * i + 1):
            print(chr(65 + j), end='')
        
        # Move to the next line
        print()

# number of rows
R = int(input("Enter the number of rows: "))
alphabet_pyramid(R)

    A
   ABC
  ABCDE
 ABCDEFG
ABCDEFGHI


### **50. Write a code to generate a square pattern using numbers.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the number pattern separated with space with the size R.

**Sample Input:**

5

**Sample Output:**

5 5 5 5 5

4 5 5 5 5

3 4 5 5 5

2 3 4 5 5

1 2 3 4 5

In [7]:
n=int(input())
k=n
for i in range(n): #rows
  p=k
  l=[]
  for j in range(i+1): #cols
    #print(p, end=" ")
    l.append(p)
    p+=1

  for j in range (i, n-1):
    #print(n, end=" ")
    l.append(n)
  k-=1
  print(*l)

5 5 5 5 5
4 5 5 5 5
3 4 5 5 5
2 3 4 5 5
1 2 3 4 5


In [12]:
def number_square_pattern(R):
    for i in range(R):
        for j in range(R):
            if j < i:
                print(i - j, end=' ')
            else:
                print(R, end=' ')
        print()

R = int(input())
print(number_square_pattern(R))

5 5 5 5 5 
1 5 5 5 5 
2 1 5 5 5 
3 2 1 5 5 
4 3 2 1 5 
None


### **51. Write a code to generate a triangle character pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=26.

#### **Output Description:**

Print the alphabet pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

    A

   A B

  A B C

 A B C D

A B C D E

In [14]:
def pattern(n):
    c = 65
    for i in range(n):
        # Print leading spaces
        for j in range(n - i - 1):
            print(' ', end='')
        
        # Print alphabet characters separated by space
        for k in range(i + 1):
            print(chr(c + k), end=' ')
        
        # Move to the next line
        print()

n = int(input())
pattern(n)

    A 
   A B 
  A B C 
 A B C D 
A B C D E 


### **52. Write a code to generate a number half pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number half pyramid pattern based on the given integer R.

**Sample Input:**

5

**Sample Output:**

1

2 9

3 8 10

4 7 11 14

5 6 12 13 15

In [45]:
n = int(input())

for i in range(n):
    for j in range(i + 1):
        x = 0
        for k in range(j):
            x = x + n - k
        if j % 2 == 0:
            print(x + i - j + 1, end=" ")
        else:
            print(x + n - i, end=" ")
    print()

1 
2 9 
3 8 10 
4 7 11 14 
5 6 12 13 15 


### **53. Generate the following pattern.**

#### **Input Description:**

Input consists of a single integer that corresponds to n, the number of rows.

where 1<=n<=100.

#### **Output Description:**

Print the character pattern from the given input n.

**Sample Input:**

5

**Sample Output:**

*****

b****

bb***

bbb**

bbbb*

In [46]:
n = int(input())

for i in range(n):
    # Print 'b' characters
    for j in range(i):
        print("b", end="")
    
    # Print '*' characters
    for j in range(i, n):
        print("*", end="")
    
    # Move to the next line
    print()

*****
b****
bb***
bbb**
bbbb*


### **54. Write a code to generate the X form of a number pattern.**

#### **Input Description:**

Given an integer N indicates X pattern.

Where 1<=N<=100.

#### **Output Description:**

Print the numbers in X form based on the given integer N.

Where 1<=N<=100

**Sample Input:**

5

**Sample Output:**

1       1

 2     2

  3   3

   4 4

    5

   4 4

  3   3

 2     2

1       1

In [56]:
# Method - 1:

def pattern(n):
    for i in range(1, n+1):
        for j in range(i-1):
            print(' ', end = "")
        print(i, end = "")
        for j in range(2*(n-i)-1):
            print(' ', end = "")            
        if i != n: print(i)
        else: print()
    for i in range(n-1, 0, -1):
        for j in range(i-1):
            print(' ', end = "")
        print(i, end = "")
        for j in range(2*(n-i)-1):
            print(' ', end = "")            
        if i != n: print(i)
        else: print()
            
n = int(input())
pattern(n)

1       1
 2     2
  3   3
   4 4
    5
   4 4
  3   3
 2     2
1       1


In [16]:
# Method - 2:

def generate_cross_pattern(N):
    count = N * 2 - 1

    for i in range(1, count + 1):
        for j in range(1, count + 1):
            if j == i or j == count - i + 1:
                print(i, end="")
            else:
                print(" ", end="")
        print()

N = int(input())
generate_cross_pattern(N)

1       1
 2     2 
  3   3  
   4 4   
    5    
   6 6   
  7   7  
 8     8 
9       9


### **55. Write a code to generate a butterfly pattern printing using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the butterfly pattern printing using stars based on the given integer R.

**Sample Input:**

5

**Sample Output:**

*        *
**      **
***    ***
****  ****
**********
**********
****  ****
***    ***
**      **
*        *

In [17]:
def generate_butterfly_pattern(R):
    # Upper part of the butterfly
    for i in range(1, R + 1):
        for j in range(1, i + 1):
            print("*", end="")
        for j in range(1, 2 * (R - i) + 1):
            print(" ", end="")
        for j in range(1, i + 1):
            print("*", end="")
        print()

    # Lower part of the butterfly
    for i in range(R, 0, -1):
        for j in range(1, i + 1):
            print("*", end="")
        for j in range(1, 2 * (R - i) + 1):
            print(" ", end="")
        for j in range(1, i + 1):
            print("*", end="")
        print()

R = int(input())
generate_butterfly_pattern(R)

*        *
**      **
***    ***
****  ****
**********
**********
****  ****
***    ***
**      **
*        *


### **56. Write a code to generate the hollow diamond inscribed in a rectangle using stars.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the hollow diamond in a rectangle using stars with the size R.

**Sample Input:**

5

**Sample Output:**

**********
****  ****
***    ***
**      **
*        *
*        *
**      **
***    ***
****  ****
**********

In [18]:
def generate_hollow_diamond(R):
    # Upper part of the diamond
    for i in range(R):
        for j in range(R - i):
            print("*", end="")
        for j in range(2 * i):
            print(" ", end="")
        for j in range(R - i):
            print("*", end="")
        print()
    
    # Lower part of the diamond
    for i in range(R):
        for j in range(i + 1):
            print("*", end="")
        for j in range(2 * (R - i - 1)):
            print(" ", end="")
        for j in range(i + 1):
            print("*", end="")
        print()

R = int(input())
generate_hollow_diamond(R)

**********
****  ****
***    ***
**      **
*        *
*        *
**      **
***    ***
****  ****
**********


### **57. Write a code to generate a matrix number pattern.**

#### **Input Description:**

Given an odd integer R indicates number of rows.R is always odd integer.

Where 1<=R<=100.

#### **Output Description:**

Print the matrix number pattern R*R based on the given integer R.

**Sample Input:**

5

**Sample Output:**

33333

32223

32123

32223

33333

In [20]:
n = int(input())

# Creating the first row
l = [str(n // 2 + 1) * n]

# Creating the subsequent rows
for i in range(1, n // 2 + 1):
    stri = ""
    var = n // 2 + 1
    
    for j in range(n // 2 + 1):
        if len(stri) >= i + 1:
            stri += stri[-1]
        else:
            stri += str(var)
            var -= 1
    
    # Concatenate the current string with its reverse (excluding last character)
    l.append(stri + stri[-2::-1])

# Duplicate and reverse the upper half to complete the pattern
l = l + l[-2::-1]

# Print each row
for i in l:
    print(i)

33333
32223
32123
32223
33333


### **58. Write a code to generate a half pyramid pattern of mulitples of the given number.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100.

#### **Output Description:**

Print the number half pyramid pattern of multiples based on the given integer R.

**Sample Input:**

10

**Sample Output:**

1

2 4

3 6 9

4 8 12 16

5 10 15 20 25

6 12 18 24 30 36

7 14 21 28 35 42 49

8 16 24 32 40 48 56 64

9 18 27 36 45 54 63 72 81

10 20 30 40 50 60 70 80 90 100

In [23]:
# Method - 1:
def pattern(n):
    for i in range(1, n+1):
        l = []
        for j in range(1, i+1):
            l.append(i*j)
        print(*l)
        
n = int(input())
pattern(n)

1
2 4
3 6 9
4 8 12 16
5 10 15 20 25
6 12 18 24 30 36
7 14 21 28 35 42 49
8 16 24 32 40 48 56 64
9 18 27 36 45 54 63 72 81
10 20 30 40 50 60 70 80 90 100


In [22]:
# Method - 2: 

def generate_multiples_pattern(R):
    # Loop through each row from 1 to R
    for i in range(1, R + 1):
        # Initialize the multiple
        multiple = i
        
        # Loop through each column in the current row
        for j in range(1, i + 1):
            # Print the current multiple
            print(multiple, end=" ")
            
            # Calculate the next multiple for the next column
            multiple += i
        
        # Move to the next line after printing each row
        print()

R = int(input())
generate_multiples_pattern(R)

1 
2 4 
3 6 9 
4 8 12 16 
5 10 15 20 25 
6 12 18 24 30 36 
7 14 21 28 35 42 49 
8 16 24 32 40 48 56 64 
9 18 27 36 45 54 63 72 81 
10 20 30 40 50 60 70 80 90 100 


### **59. Write a code to generate a aplhabet half pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=26.

#### **Output Description:**

Print the alphabet half pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

EEEEE

 DDDD

  CCC

   BB

    A

In [24]:
def generate_alphabet_pattern(R):
    # Convert R to a character for starting alphabet
    start_char = chr(64 + R)
    
    # Loop through each row from 1 to R
    for i in range(R, 0, -1):
        # Print spaces before the characters
        print(" " * (R - i) + (start_char * i))
        # Decrement the ASCII value for next character
        start_char = chr(ord(start_char) - 1)

R = int(input())
if 1 <= R <= 26:
    generate_alphabet_pattern(R)
else:
    print("Invalid input. Please enter a number between 1 and 26.")

EEEEE
 DDDD
  CCC
   BB
    A


### **60. Write a code to generate a pyramid of numbers and aplhabets.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=100

#### **Output Description:**

Print the number half pyramid pattern of multiples based on the given integer R.

**Sample Input:**

8

**Sample Output:**

       A1

      AB12

     ABC123

    ABCD1234

   ABCDE12345

  ABCDEF123456

 ABCDEFG1234567

ABCDEFGH12345678

In [25]:
def generate_pyramid_pattern(n):
    for i in range(n):  # Loop through each row
        # Print leading spaces
        for j in range(i, n - 1):
            print(" ", end="")

        # Print alphabets
        for j in range(i + 1):
            print(chr(65 + j), end="")

        # Print numbers
        for j in range(1, i + 2):
            print(j, end="")

        # Move to the next line after printing each row
        print()

n = int(input("Enter the number of rows (1-100): "))
if 1 <= n <= 100:
    generate_pyramid_pattern(n)
else:
    print("Invalid input. Please enter a number between 1 and 100.")

       A1
      AB12
     ABC123
    ABCD1234
   ABCDE12345
  ABCDEF123456
 ABCDEFG1234567
ABCDEFGH12345678


### **61. Write a code to generate a alphabet pyramid pattern.**

#### **Input Description:**

Given an integer R indicates number of rows.

Where 1<=R<=13.

#### **Output Description:**

Print the alphabet pyramid pattern according to the given integer R.

**Sample Input:**

5

**Sample Output:**

    A

   CCC

  EEEEE

 GGGGGGG

IIIIIIIII

In [26]:
def pattern(n):
    c = 65
    for i in range(1, n+1):
        for j in range(n-i):
            print(' ', end = "")
        for j in range(2*i - 1):
            print(chr(c), end = "")
        print()
        c += 2

n = int(input())
pattern(n)

    A
   CCC
  EEEEE
 GGGGGGG
IIIIIIIII
